vaadin-upload.html

```diff
@@ -0,0 +1,225 @@
+<link rel="import" href="color.html">
+<link rel="import" href="font-icons.html">
+<link rel="import" href="typography.html">
+<link rel="import" href="vaadin-button.html">
+<link rel="import" href="vaadin-progress-bar.html">
+
+<dom-module id="material-upload" theme-for="vaadin-upload">
+  <template>
+    <style>
+      :host(:not([nodrop])) {
+        overflow: hidden;
+        border: 1px dashed var(--material-divider-color);
+        border-radius: 4px;
+        padding: 8px 16px;
+        transition: border-color 0.6s;
+        position: relative;
+      }
+
+      [part="primary-buttons"] {
+        display: inline-flex;
+        flex-wrap: wrap;
+        align-items: baseline;
+      }
+
+      /* TODO(jouni): unsupported selector (not sure why there's #addFiles element wrapping the upload button) */
+      [part="primary-buttons"] > * {
+        display: block;
+        flex-grow: 1;
+      }
+
+      [part="upload-button"] {
+        display: block;
+        margin: 0 -8px;
+      }
+
+      [part="drop-label"] {
+        text-align: center;
+        white-space: normal;
+        padding: 0 24px;
+        color: var(--material-secondary-text-color);
+        font-family: var(--material-font-family);
+        font-size: var(--material-small-font-size);
+      }
+
+      :host([dragover-valid]) {
+        border-color: var(--material-primary-color);
+        transition: border-color 0.1s;
+      }
+
+      :host([dragover-valid]) [part="drop-label"] {
+        color: var(--material-primary-text-color);
+      }
+
+      [part="drop-label-icon"] {
+        display: inline-block;
+        margin-right: 8px;
+      }
+
+      [part="drop-label-icon"]::before {
+        content: var(--material-icons-upload);
+        font-family: material-icons;
+        font-size: var(--material-icon-font-size);
+        line-height: 1;
+      }
+
+      /* Ripple */
+
+      :host::before {
+        content: "";
+        position: absolute;
+        width: 100px;
+        height: 100px;
+        border-radius: 50%;
+        top: 50%;
+        left: 50%;
+        pointer-events: none;
+        background-color: var(--material-primary-color);
+        opacity: 0;
+        transform: translate(-50%, -50%) scale(0);
+        transition: transform 0s cubic-bezier(0.05, 0.8, 0.5, 1), opacity 0.4s linear;
+        transition-delay: 0.4s, 0s;
+      }
+
+      :host([dragover-valid])::before {
+        transform: translate(-50%, -50%) scale(6);
+        opacity: 0.1;
+        transition-duration: 2s, 0.1s;
+        transition-delay: 0s, 0s;
+      }
+    </style>
+  </template>
+</dom-module>
+
+<dom-module id="material-upload-file" theme-for="vaadin-upload-file">
+  <template>
+    <style>
+      :host {
+        padding: 8px 0;
+      }
+
+      :host(:not(:first-child)) {
+        border-top: 1px solid var(--material-divider-color);
```

Comment:
Looks like a grid of files with these lines.

Two-line lists in the guide, typically, don’t use horizontal borders between items: https://material.io/design/components/lists.html#types

Let's consider removing them maybe?

Reply:
Good point. Removed the dividers.